### PR TITLE
Export config

### PR DESCRIPTION
**Pull request by FionaMorgan**
Function .api.exportconfig and .api.torqnamespaces added to allow for values of non-default namespace variables to be seen. Outputs a table of name, value and description and is called using ```.api.exportconfig[.api.torqnamespaces]``` or with any namespace arguments (e.g. `.usage). An additional function .api.exportallconfig has been added as an automatic ```.api.exportconfig[.api.torqnamespaces]``` call

### CODE REVIEW

File: code/common/api.q [hunk 1]
```diff
@@ -72,6 +72,21 @@ search:{[s;c]
 // search function
 s:.api.search[;0b]
 
+// input list of namespaces for exportconfig
+KEYS:`$".",'string each key[`] except `$'.Q.an;
```
Comment (ThomasSmyth) [inline code/common/api.q]: Work with symbols?
```
` sv'`,'key[`]except`$'.Q.an
```

File: code/common/api.q [hunk 2]
```diff
@@ -72,6 +72,21 @@ search:{[s;c]
...
+
+// export the current state of config variables, takes in list of symbols of namespaces (e.g. `.usage`.procs)
+exportconfig:{
+  // selects only variables in inpoutted namespaces
+  :?[.api.f`;
+    ((=;`vartype;enlist`variable);(in;`namespace;enlist x));
+    {x!x}enlist`name;
+  // returns name, value and description
+    `val`descrip!(((';.:);`name);`descrip)
```
Comment (ThomasSmyth) [inline code/common/api.q]: Whats `(;;,:)` supposed to be? A good tip is that you don't want to see q keywords transformed in parse tree you can treat them as symbols instead:
```
q)parse"select `count i from tab"
?
`tab
()
0b
(,`x)!,(,`count;`i)
```

File: code/common/grafana.q [hunk 3]
```diff
@@ -0,0 +1,150 @@
+\d .grafana
```
Comment (ThomasSmyth) [inline code/common/grafana.q]: Should this file be in here?

File: docs/utilities.md [hunk 4]
```diff
@@ -1190,4 +1190,3 @@ API Table
 |.gc.run|function|.gc|Run garbage collection, print debug info before and after|||
 |.mem.objsize|function|.mem|Returns the calculated memory size in bytes used by an object.  It may take a little bit of time for objects with lots of nested structures (e.g. lots of nested columns)|[q object]|size of the object in bytes|
 |.tplog.check|function|.tplog|Checks if tickerplant log can be replayed.  If it can or can replay the first X messages, then returns the log handle, else it will read log as byte stream and create a good log and then return the good log handle |[logfile (symbol), handle to the log file to check; lastmsgtoreplay (long), the index of the last message to be replayed from log ]|handle to log file, will be either the input log handle or handle to repaired log, depends on whether the log was corrupt|
-
```
Comment (ThomasSmyth) [inline docs/utilities.md]: Can remove changes here too.

Comment (FionaMorgan) [reply docs/utilities.md]: Does this mean add a single line?

Comment (ThomasSmyth) [reply docs/utilities.md]: Just checkout the original version from master:
```
git checkout master -- <path to file>
```

File: code/common/api.q [hunk 5]
```diff
@@ -72,6 +72,21 @@ search:{[s;c]
 // search function
 s:.api.search[;0b]
 
+// input list of namespaces for exportconfig
+KEYS:` sv'`,'key[`]except`$'.Q.an;
```
Comment (jonnypress) [inline code/common/api.q]: Can you give this a better name like "torqnamespaces" or something.  Keys is a bit vague

File: code/common/api.q [hunk 6]
```diff
@@ -72,6 +72,21 @@ search:{[s;c]
...
+exportconfig:{
+  // selects only variables in inpoutted namespaces
+  :?[.api.f`;
+    ((=;`vartype;enlist`variable);(in;`namespace;enlist x));
+    {x!x}enlist`name;
+  // returns name, value and description
+    `val`descrip!((value';`name);`descrip)
+   ];
+ }
+
```
Comment (jonnypress) [inline code/common/api.q]: Looks good, though can you also add exportallconfig as 

exportallconfig:{exportconfig[torqnamespaces]}

**Comment by jonnypress** [comment]
Can you add brief documentation in code/common/apidetails.q?

File: code/common/apidetails.q [hunk 7]
```diff
@@ -13,6 +13,8 @@ add[`.api.find;1b;"Generic method for finding functions/variables/tables/views.
 add[`.api.search;1b;"Generic method for searching all function definitions for a specific string. s is based on this";"[string: search string; boolean: whether the search is context senstive";"table of matching functions and definitions"]
 add[`.api.add;1b;"Add a function to the api description table";"[symbol:the name of the function; boolean:whether it should be called externally; string:the description; dict or string:the parameters for the function;string: what the function returns]";"null"]
 add[`.api.fullapi;1b;"Return the full function api table";"[]";"api table"]
+add[`.api.exportconfig;1b;"Return value table of requested torq variables and descriptions";"[result of .api.f]";"keyed table of name, value, description"]
```
Comment (jonnypress) [inline code/common/apidetails.q]: The input is the list of namespaces to return isn't it? 
e.g. .api.exportconfig[`.proc`.servers] 
? 
The description is a bit confusing

Comment (FionaMorgan) [reply code/common/apidetails.q]: That's correct, it's the namespace. Will change

**Pull request merged**